src/test/java/spoon/test/serializable/SourcePositionTest.java

```diff
@@ -71,4 +76,18 @@ public void testSourcePosition() throws IOException {
 		CtField<?> elem2 = typeFromFile.getField("a");
 		assertTrue(elem1.getPosition().getFile().equals(elem2.getPosition().getFile()));
 	}
+
+	@Test
+	public void test_sourcePositionOfTypeInFieldExists() {
+		// contract: the type reference of type of the field should have a source position
+		final Launcher launcher = new Launcher();
+		launcher.addInputResource("src/test/resources/spoon/test/sourcePosition/FieldType.java");
+		launcher.addInputResource("src/test/resources/spoon/test/sourcePosition/SourcePartitionValidator.java");
+		CtModel model = launcher.buildModel();
+
+		CtField field = (CtField) model.getElements(
+				element -> element instanceof CtField &&
+						((CtField) element).getSimpleName().equals("pleaseAttachSourcePositionToMyType")).stream().findFirst().get();
+		assertThat(field.getType().getPosition(), not(instanceOf(NoSourcePosition.class)));
```

Comment:
Can`t you check here with https://spoon.gforge.inria.fr/mvnsites/spoon-core/apidocs/spoon/reflect/cu/SourcePosition.html#isValidPosition()?

Reply:
Oh, right. I wasn't aware of this API method. Thanks!